src/meshapi/views/forms.py

```diff
@@ -35,8 +37,25 @@ class JoinFormRequest:
     ncl: bool
 
 
+def acquire_lock_for_queryset(queryset: QuerySet, object_count=None):
+    # This code may look useless but is extremely important. Here we use select_for_update()
+    # to acquire locks on every single row in the given table. This is a bit hacky, but based
```

Comment:
Fix me: this isn’t alway every row anymore

Reply:
What are the implications of that?